File: test-project/greeter.provider.test/src/test/java/tests/GreeterTest.java
package tests;

import examples.greeter.api.Greeter;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;

public class GreeterTest {
    @Test
    public void testLocate() {
        Greeter greeter = new GreeterLocator().findGreeter();
        assertFalse(greeter.hello().isBlank());
    }
}

File: test-project-kotlin/greeter.provider.test/src/test/kotlin/tests/GreeterTest.kt
package tests

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.assertFalse

class GreeterTest {
    @Test
    fun testLocate() {
        val greeter = GreeterLocator().findGreeter()
        assertFalse(greeter.hello().isBlank())
    }

    @Test
    fun testGeneratedResource() {
        val resource = object: Any() {}.javaClass.getResourceAsStream("/generated-resource.txt")
        if (resource == null) {
            throw RuntimeException("Couldn't load generated resource")
        }
    }
}
